near/omni-types/src/prover_types.rs

```diff
@@ -0,0 +1,41 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::U128;
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::AccountId;
+
+use crate::{OmniAddress, TransferMessage};
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone)]
+pub struct InitTransferMessage {
+    pub transfer: TransferMessage,
+    pub contract: OmniAddress,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone)]
+pub struct FinTransferMessage {
+    pub nonce: U128,
+    pub claim_recipient: AccountId,
+    pub contract: OmniAddress,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone)]
+pub struct DeployTokenMessage {
+    pub token: AccountId,
+    pub token_address: OmniAddress,
+    pub contract: OmniAddress,
+}
+
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Debug, Clone)]
+pub enum ProofResult {
```

Comment:
I think another ProofResult still exists in the omni-types package. Do we want to remove that one?

Reply:
5e7b9641662e6e3cf33a73dd75102c8b474cb35a